src/main/java/com/shc/silenceengine/graphics/cameras/FPSCamera.java

```diff
@@ -139,13 +140,20 @@ public FPSCamera moveDown(float amount)
 
     public FPSCamera rotateX(float angle)
     {
+        // If the current angle is already on the limit, do nothing.
+        if (angleX <= -ANGLE_LIMIT_X || angleX >= ANGLE_LIMIT_X) {
```

Comment:
You didn't get this right, the limit means the angle can be inside -angle to + angle, not outside the range.